services/hack/joinHack.js

```diff
@@ -17,31 +17,30 @@ export const join = async (event, context) => {
       { $push: { team: mongUserID } },
       { new: true, upsert: true }
     ).populate("team", "-email", User);
-    // const doc = await Query.find()
-    // const users = await Team.find();
-    // for(const team_key in res.team){
-    //   users.forEach(member => {
-    //     if(JSON.stringify(member._id) == JSON.stringify(res.team[team_key])){
-    //       res.team[team_key] = {_id: member._id, name: member.name};
-    //     }
-    //   });
-    // }
+    //console.log(res);
+
+    if (res.status === "Submitted") throw new Error("Unable to join");
+
+    if (res.status === "Canceled") throw new Error("Hack is canceld");
+
+    if (res.status === "Team Closed") throw new Error("Hack is closed");
```

Comment:
Just a note on this, We should look at disabling the join button with these statuses

Reply:
one other thing to clean this up a little, something that Jo showed me.
you could try
`if(["Submitted", "Canceled", "Team Closed"].includes(res.status)) throw new Error("Unable to join hack due to status")`
just cleans those three lines up into one

Reply:
We will change the logic a bit on the front end to disable the button, which shouldn't be a problem.